src/ui_basic/dropdown.h

```diff
@@ -433,6 +407,63 @@ template <typename Entry> class Dropdown : public BaseDropdown {
 		filtered_entries.push_back(std::unique_ptr<Entry>(new Entry(value)));
 		BaseDropdown::add(name, size(), pic, select_this, tooltip_text, hotkey);
 	}
+	bool check_hotkey_match(const std::string& input_text) {
+		for (auto& x : unfiltered_entries) {
+			if (input_text == to_lower(x.hotkey) && is_in_filtered_list(x.value)) {
+				if (x.onShortcutMatch) {
+					x.onShortcutMatch(x.value);
+				} else {
+					log_err("hotkey match: %s but no hotkey function available!", x.hotkey.c_str());
```

Comment:
This happens when pressing P in the editor with the tool dropdown open. So perhaps make this just an inobtrusive reminder to add this feature there as well:
```suggestion
					verb_log_dbg("hotkey match: %s but no hotkey function available!", x.hotkey.c_str());
```